### PR TITLE
Weather close search on click somewhere on body

### PR DESCRIPTION
**Pull request by uplne**
1. User should be able to close the autocomplete tool by clicking on X or by clicking anywhere outside the list.
   xcbpcj1dei
2. This PR should also remove the Error retrieving city data which was introduced last week.

**Adding fastdom will be the next PR after this and scala template PRs will be out. I am already expecting many conflicts so I don't want to do too many things at the same time.**

### CODE REVIEW

File: static/src/javascripts/projects/facia/modules/onwards/search-tool.js [hunk 1]
```diff
@@ -45,11 +48,49 @@ define([
...
-                e.preventDefault();
+                var isInput = $(e.target).hasClass('js-search-tool-input'),
+                    isLink  = this.isLink(e.target);
+
+                if (isInput) {
+                    e.preventDefault();
+                    mediator.emit('autocomplete:toggle', true);
+                } else if (isLink) {
+                    e.preventDefault();
+                    $('.active').removeClass('active');
```
Comment (piuccio) [inline static/src/javascripts/projects/facia/modules/onwards/search-tool.js]: Is this removing the `.active` class on the whole document? Maybe it should be something like `$(weatherWidget).find('.active').removeClass('active')` for better isolation.
Or you could keep a reference of the active element, so you don't have to query the DOM to get it.

**Comment by michaelwmcnamara** [comment]
To be tested on mobile devices also.
Specifically 
- what happens if you close keyboard 
- what happens if you tap on page outside of dropdown (more relevant for tablets than phones)

**Comment by uplne** [comment]
@michaelwmcnamara I checked it on all devices. There is not really anything major on them but I wrote down few points. Some of it we already have in that awesome excel sheet you've created.

I will focus on fixing those in the separate pull requests.

**Comment by phamann** [comment]
:+1: LGTM!

**Pull request merged**